test/Broadway/EventSourcing/AggregateFactory/ReflectionAggregateFactoryTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Broadway\EventSourcing\AggregateFactory;
+
+use Broadway\Domain\DomainEventStream;
+use Broadway\EventSourcing\EventSourcedAggregateRoot;
+use Broadway\TestCase;
+
+/**
+ *
+ */
+final class ReflectionAggregateFactoryTest extends TestCase
+{
+    /**
+     * @var ReflectionAggregateFactory
+     */
+    private $factory;
+
+    public function setUp()
+    {
+        $this->factory = new ReflectionAggregateFactory();
+    }
+
+    /**
+     * @test
+     */
+    public function it_creates_instance_of_aggregate_with_private_constructor()
+    {
+        $aggregate = $this->factory->create(
+            MyTestAggregateRootForReflectionFactory::class,
+            new DomainEventStream([])
+        );
+
+        $this->assertInstanceOf(MyTestAggregateRootForReflectionFactory::class, $aggregate);
+    }
+
+    /**
+     * @test
+     * @expectedException \LogicException
+     * @expectedExceptionMessage Impossible to initialize "stdClass"
+     */
+    public function it_does_not_handle_weird_classes()
+    {
+        $this->factory->create(\stdClass::class, new DomainEventStream([]));
+    }
+}
+
+// TODO: separate folder with test aggregates?
```

Comment:
Yeah, that is a risk indeed, Although I think it won't be that big of a deal.
Most of these dummies are quit specific for a testcase so can have a very specific name.

Let's at least not try to solve that in this PR. Feel free to open a new PR if you want to :-)